game/world/objects/npc.cpp

```diff
@@ -1765,6 +1766,8 @@ void Npc::takeDamage(Npc& other, const Bullet* b, const CollideMask bMask, int32
         }
       else if(isDead()) {
         owner.sendPassivePerc(*this,other,*this,PERC_ASSESSMURDER);
+        // Register monsters for respawn (method handles human / monster / orc evaluation)
+        RespawnObject::registerObject(size_t(instanceSymbol()), hnpc->wp.c_str(), guild());
```

Comment:
This might break in swampshark in Addon:
```
instance MIS_Addon_Swampshark_Lou(Mst_Default_Swampshark) <--- normal shark
{
	B_SetVisuals_Swampshark();
	Npc_SetToFistMode(self);
	CreateInvItems(self,ITKE_Addon_Tavern_01,1);  <--- Quest item!
};
```
Again, see `WorldObjects::resetPositionToTA`/`Npc::resetPositionToTA` - body cleanup handle there